File: chroma-league/src/main/java/net/booone/chromaleague/hud/parts/resource/RenektonFuryBar.java
package net.booone.chromaleague.hud.parts.resource;

import net.booone.chromaleague.hud.parts.ProgressBar;
import net.booone.chromaleague.state.GameStateHelper;
import net.booone.razersdk.animation.AnimatedFrame;
import net.booone.razersdk.animation.Frame;
import net.booone.razersdk.color.StaticColor;

public class RenektonFuryBar extends AnimatedFrame {
    @Override
    public Frame getFrame() {
        final int resourcePercentage = GameStateHelper.getResourcePercentage();
        if (resourcePercentage < 50) {
            addAnimationFrame(new ProgressBar(ResourceBars.getResourceBarKeys(),
                    resourcePercentage, StaticColor.WHITE));
        } else {
            addAnimationFrame(new ProgressBar(ResourceBars.getResourceBarKeys(),
                    resourcePercentage, StaticColor.RED));
        }
        return super.getFrame();
    }
}

File: src/hud/parts/resource/RenektonFuryBar.ts
import {AnimatedFrame} from '../../../razer-sdk/animation/AnimatedFrame.js';
import {Frame} from '../../../razer-sdk/animation/Frame.js';
import {StaticColor} from '../../../razer-sdk/color/StaticColor.js';
import {ProgressBar} from '../ProgressBar.js';
import {getResourceBarKeys} from './ResourceBars.js';
import {getResourcePercentage} from '../../../state/GameStateHelper.js';

export class RenektonFuryBar extends AnimatedFrame {
    pollFrame(): Frame {
        const resourcePercentage = getResourcePercentage();
        if (resourcePercentage < 50) {
            this.addAnimationFrame(new ProgressBar(getResourceBarKeys(), resourcePercentage, StaticColor.WHITE));
        } else {
            this.addAnimationFrame(new ProgressBar(getResourceBarKeys(), resourcePercentage, StaticColor.RED));
        }
        return super.pollFrame();
    }
}
